src/components/PupilComponents/QuizMatchAnswer/QuizMatchAnswer.tsx

```
@@ -53,19 +54,72 @@ export const QuizMatchAnswer = () => {
     }),
   );
 
-  const matchItems: { id: string; label: JSX.Element }[] = [];
-  const choiceItems: { id: string; label: JSX.Element }[] = [];
-
-  Object.entries(answers).forEach(([key, value], index) => {
-    matchItems.push({
-      id: index.toString(),
-      label: <MathJaxWrap key={`match-${index}`}>{key}</MathJaxWrap>,
-    });
-    choiceItems.push({
-      id: index.toString(),
-      label: <MathJaxWrap key={`choice-${index}`}>{value}</MathJaxWrap>,
+  const matchItems: { id: string; label: JSX.Element; announcement: string }[] =
+    useMemo(() => {
+      const matchItems: {
+        id: string;
+        label: JSX.Element;
+        announcement: string;
+      }[] = [];
+      Object.keys(answers).forEach((key, index) => {
+        matchItems.push({
+          id: index.toString(),
+          label: <MathJaxWrap key={`match-${index}`}>{key}</MathJaxWrap>,
+          announcement: key,
+        });
+      });
+      return matchItems;
+    }, [answers]);
+
+  const choiceItems: {
+    id: string;
+    label: JSX.Element;
+    announcement: string;
+  }[] = useMemo(() => {
+    const choiceItems: {
+      id: string;
+      label: JSX.Element;
+      announcement: string;
+    }[] = [];
+    Object.values(answers).forEach((value, index) => {
+      choiceItems.push({
+        id: index.toString(),
+        label: <MathJaxWrap key={`choice-${index}`}>{value}</MathJaxWrap>,
+        announcement: value,
+      });
     });
-  });
+    return choiceItems;
+  }, [answers]);
+
+  useEffect(() => {
+    setDocumentLoaded(true);
+    if (documentLoaded) {
+      choiceItems.forEach((item) => {
+        const element = document.getElementById(
+          `oak-quiz-match-item-${item.id}`,
+        );
+        if (element) {
+          const ariaLabel =
```

is there another way to query this aria label than digging through children? what will happen if someone will change markup at any point?

---

Yeah I think this is an easy thing to miss as well if the structure changes, as it will fail silently I think. (same on line 115)

something like 

`element?.querySelector(".class")?.ariaLabel;`

or 

`element?.querySelector("[aria-label]")?.ariaLabel;` If it is the first nested aria-label.

---

thanks both; i agree i think theres a few ways to make it a bit safer.

---

would you be happier with this method: 
element?.querySelector("[aria-label]")?.ariaLabel;